src/lib/modals/Modal.svelte

```diff
@@ -114,17 +107,23 @@
 </script>
 
 {#if open}
+  <!-- backdrop -->
+  <div class={classNames('fixed inset-0 z-40', backdropClasses)} />
+  <!-- dialog -->
   <div
-    tabindex="-1"
-    class={mainClass}
-    aria-modal="true"
-    role="dialog"
     on:keydown|preventDefault={handleKeys}
     on:wheel|preventDefault
     use:prepareFocus
     use:focusTrap
-    on:click={autoclose ? onAutoClose : null}>
-    <div class="flex p-4 w-full {sizes[size]} h-full md:h-auto max-h-screen">
+    on:click={autoclose ? onAutoClose : null}
+    class={classNames(
+      'fixed top-0 left-0 right-0 h-modal md:inset-0 md:h-full z-50 w-full p-4 flex',
+      ...getPlacementClasses()
+    )}
+    tabindex="-1"
+    aria-modal="true"
+    role="dialog">
+    <div class="relative {sizes[size]} max-h-full md:h-auto flex">
```

Comment:
I am wondering if we have to do a `md:h-auto` (height should always be auto right? Even on a small screen since we have `max-h-full`)

Reply:
Rest of the changes look good to me. Thank you for the quick fix